pkg/plugin/linuxutil/linuxutil_linux.go

```diff
@@ -11,15 +11,19 @@ import (
 	"time"
 
 	hubblev1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	lru "github.com/hashicorp/golang-lru/v2"
 	kcfg "github.com/microsoft/retina/pkg/config"
 	"github.com/microsoft/retina/pkg/log"
 	"github.com/microsoft/retina/pkg/plugin/registry"
+	"github.com/pkg/errors"
 	"github.com/safchain/ethtool"
 	"go.uber.org/zap"
 )
 
 const defaultLimit = 2000
 
+var globalLruCache *lru.Cache[string, struct{}]
```

Comment:
Don't use global variable, inject the cache to the functions. Something like
```
unsupportedInterfacesCache, err = lru.New[string, struct{}](int(defaultLimit))
...
ethReader := NewEthtoolReader(ethtoolOpts, ethHandle, unsupportedInterfacesCache)
...
```

Decide on the cache name if it contains something other than unsupported interfaces.